flink-connector-prometheus-request-signer-amp/src/main/java/org/apache/flink/connector/prometheus/sink/aws/AmazonManagedPrometheusWriteRequestSigner.java

```diff
@@ -0,0 +1,86 @@
+package org.apache.flink.connector.prometheus.sink.aws;
+
+import org.apache.flink.connector.prometheus.sink.PrometheusRequestSigner;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.commons.lang3.StringUtils;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+/** Sign a Remote-Write request to Amazon Managed Service for Prometheus (AMP). */
+public class AmazonManagedPrometheusWriteRequestSigner implements PrometheusRequestSigner {
+    // Header names
+    private static final String X_AMZ_CONTENT_SHA_256 = "x-amz-content-sha256";
+    private static final String AUTHORIZATION = "Authorization";
+
+    private final URL remoteWriteUrl;
+    private final String awsRegion;
+
+    /**
+     * Creates a signer instance using DefaultAWSCredentialsProviderChain.
+     *
+     * @param remoteWriteUrl URL of the remote-write endpoint
+     * @param awsRegion Region of the AMP workspace
+     */
+    public AmazonManagedPrometheusWriteRequestSigner(String remoteWriteUrl, String awsRegion) {
+        Preconditions.checkArgument(
+                StringUtils.isNotBlank(awsRegion), "Missing or blank AMP workspace region");
+        Preconditions.checkNotNull(
```

Comment:
I think this is supposed to be `Preconditions.checkArgument(` . We should also have unit tests for this.